streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -293,6 +293,11 @@ boolean tryToCompleteRestoration() {
      * @throws TaskMigratedException if the task producer got fenced (EOS only)
      */
     void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
+        if (activeTaskIds().isEmpty()) {
```

Comment:
@guozhangwang @abbccdda We are still at risk of the race condition on a `TaskMigratedException` where we call `onRevoked` instead of `onLost` -- when we directly call `handleLostAll` we'll remove all active tasks, then invoking `handleRevocation` will throw `IllegalStateException` below

Reply:
I've just discussed this with @abbccdda when we were discussing another PR: https://github.com/apache/kafka/pull/8168 and it turns out that in `handleRevocation` we should actually not throw IllegalStateException at all, since with KIP-429 the `handleAssignment` triggered from the streams-assignor should close the tasks already so when `handleRevocation` is triggered we would not have those tasks to be suspended, so we should actually remove that throw statement and replaced with a info log only.